chain-signatures/contract-eth/contracts/ChainSignatures.sol

```diff
@@ -1,170 +1,111 @@
+        string params;
     }
 
     struct AffinePoint {
         uint256 x;
         uint256 y;
     }
 
-    uint256 public threshold;
-    mapping(bytes32 => SignatureRequest) public pendingRequests;
-    uint256 public requestCounter;
-    PublicKey public publicKey;
-
-    mapping(bytes32 => uint256) public depositToRefund;
-
-    event SignatureRequested(bytes32 indexed requestId, address requester, uint256 epsilon, uint256 payloadHash, string path);
-    event SignatureResponded(bytes32 indexed requestId, SignatureResponse response);
-
-    /// @custom:oz-upgrades-unsafe-allow constructor
-    constructor() initializer {}
-
-    function initialize(PublicKey memory _publicKey) public initializer {
-        publicKey = _publicKey;
+    struct Signature {
+        AffinePoint bigR;
+        uint256 s;
+        uint8 recoveryId;
     }
 
-    function getPublicKey() public view returns (PublicKey memory) {
-        return publicKey;
+    struct Response {
+        bytes32 requestId;
+        Signature response;
     }
 
-    function derivedPublicKey(string memory path, address _predecessor) public view returns (PublicKey memory) {
-        address predecessor = _predecessor == address(0) ? msg.sender : _predecessor;
-        uint256 epsilon = deriveEpsilon(path, predecessor);
-        PublicKey memory _derivedPublicKey = deriveKey(publicKey, epsilon);
-        return _derivedPublicKey;
+    uint256 signatureDeposit;
+
+    event SignatureRequested(
+        address sender,
+        bytes32 payload,
+        string path,
+        uint32 keyVersion,
+        string algo,
+        string dest,
+        string params,
+        uint256 deposit,
+        uint256 chainId,
+        uint256 timestamp
+    );
+
+    event SignatureResponded(
+        bytes32 indexed requestId,
+        address responder,
+        Signature response
+    );
+
+    event Withdraw(address indexed owner, uint amount);
+
+    constructor(address _mpc_network, uint256 _signatureDeposit) {
+        _grantRole(DEFAULT_ADMIN_ROLE, _mpc_network);
+        signatureDeposit = _signatureDeposit;
     }
 
-    function deriveKey(PublicKey memory _publicKey, uint256 epsilon) public pure returns (PublicKey memory) {
-        // G * epsilon + publicKey
-        (uint256 epsilonGx, uint256 epsilonGy) = Secp256k1.ecMul(epsilon, Secp256k1.GX, Secp256k1.GY);
-        (uint256 resultX, uint256 resultY) = Secp256k1.ecAdd(epsilonGx, epsilonGy, _publicKey.x, _publicKey.y);
-        return PublicKey(resultX, resultY);
+    function sign(
+        SignRequest memory _request
+    ) external payable {
+        require(msg.value >= signatureDeposit, "Insufficient deposit");
+
+        emit SignatureRequested(
+            msg.sender,
+            _request.payload,
+            _request.path,
+            _request.keyVersion,
+            _request.algo,
+            _request.dest,
+            _request.params,
+            msg.value,
+            block.chainid,
+            block.timestamp
```

Comment:
Do we want to keep this field? Do we want to include it in `requestId`?
Discussion: https://sig-tar2858.slack.com/archives/C07SKEDJ8QL/p1739280405646959?thread_ts=1739204631.938849&cid=C07SKEDJ8QL